db/db_impl_write.cc

```diff
@@ -62,9 +62,20 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
                          WriteBatch* my_batch, WriteCallback* callback,
                          uint64_t* log_used, uint64_t log_ref,
                          bool disable_memtable) {
+  const bool lock_memtable = !disable_memtable;
+  std::unique_ptr<InstrumentedMutexLock> serial_memtable_guarantee;
+  if (lock_memtable) {
+    serial_memtable_guarantee = std::unique_ptr<InstrumentedMutexLock>(
+        new InstrumentedMutexLock(&mem_mutex_));
+  }
+  const bool batch_writes = false;
```

Comment:
Yeah I am thinking something like options.disable_group_batch and options.enable_wal_buffer